### PR TITLE
Introducing VoiceSearch

### PR DESCRIPTION
**Pull request by andrenatal**
Ok, this is ready to go. 

@keianhzo r?

### CODE REVIEW

**Comment by andrenatal** [comment]
This is the SDK's repo: https://github.com/mozilla/androidspeech

File: app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationURLBar.java [hunk 1]
```diff
@@ -47,6 +59,8 @@
     private int mURLProtocolColor;
     private int mURLWebsiteColor;
     private Pattern mURLPattern;
+    private MozillaSpeechService mMozillaSpeechService;
+    private Context mContext;
```
Comment (keianhzo) [inline app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationURLBar.java]: you can use getContext() to get the context from any widget

File: app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationBarWidget.java [hunk 2]
```diff
@@ -238,6 +238,8 @@ public void onClick(View view) {
                 mFocusEnterButton, mFocusExitButton, mResizeEnterButton, mResizeExitButton,
                 mPreset0, mPreset1, mPreset2, mPreset3));
 
+        mURLBar.setContext(aContext);
```
Comment (keianhzo) [inline app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationBarWidget.java]: Can be avoided using getContext() from NavigationURLBar

File: app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationURLBar.java [hunk 3]
```diff
@@ -77,6 +91,8 @@ public void onFocusChange(View view, boolean b) {
             }
         });
         mURL.addTextChangedListener(mURLTextWatcher);
+        mMozillaSpeechService = MozillaSpeechService.getInstance();
+        mMozillaSpeechService.addListener(mVoiceSearchListener);
```
Comment (keianhzo) [inline app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationURLBar.java]: Unregister that listener on:

```
@Override
public void releaseWidget() {
}
```

**Comment by keianhzo** [comment]
Looks good, a couple of minor things. I'll fix those when I finish the implementation.

**Comment by keianhzo** [comment]
Looks good @andrenatal just a couple of minor things. I'll fix those when I finish the implementation.

**Pull request merged**